src/fuzzer.py

```diff
@@ -186,30 +187,92 @@ def store_test_case(self, test_case: TestCase, require_retires: bool):
+
+        The test cases and inputs are saved to individual files in the
+        user-specified working directory (or the user's current directory, if no
+        directory is specified).
+
+        Optionally, this accepts an 'input_format' string that corresponds to
+        one of the supported modes in the Input class' 'save()' method. If left
+        as None, a default will be used.
+
+        This also accepts 'permit_overwrite', which allows for an existing set
+        of generated files to be overwritten in the same directory. By default
+        this is false.
+        """
         LOGGER.fuzzer_start(0, datetime.today())
 
         # prepare for generation
         STAT.test_cases = num_test_cases
-        CONF.program_generator_seed = program_generator_seed
-        program_gen = factory.get_generator(self.instruction_set)
-        input_gen = factory.get_input_generator()
 
-        # generate test cases
-        Path(self.work_dir).mkdir(exist_ok=True)
-        for i in range(0, num_test_cases):
-            test_case_dir = self.work_dir + "/tc" + str(i)
+        # log the given parameters
+        log_msg = "Generating batch of %d program(s) and %d input(s). " \
+                  "(program seed: %d) (input seed: %d)" % \
+                  (num_test_cases, num_inputs, program_seed, input_seed)
+        if input_format:
+            log_msg += " (input file format: %s)" % input_format
+        LOGGER.inform("fuzzer", log_msg)
+
+        # if no working directory was supplied, use the current directory
+        out_dir = self.work_dir
+        if not out_dir or out_dir == "":
+            out_dir = os.getcwd()
+        Path(out_dir).mkdir(exist_ok=True, mode=0o755)
+
+        # create the two generators
+        self.generator = factory.get_generator(self.instruction_set)
+        self.input_gen = factory.get_input_generator()
+
+        # invoke the test-case generator to create assembly files
+        for i in range(num_test_cases):
+            # attempt to create a directory for the test case
+            test_case_dir = out_dir + "/tc" + str(i)
             try:
-                Path(test_case_dir).mkdir(exist_ok=permit_overwrite)
+                Path(test_case_dir).mkdir(exist_ok=permit_overwrite, mode=0o755)
             except FileExistsError:
                 LOGGER.error(f"Directory '{test_case_dir}' already exists\n"
-                             "       Use --permit-overwrite to overwrite the test case")
-
-            program_gen.create_test_case(test_case_dir + "/" + "program.asm", True)
-            inputs = input_gen.generate(CONF.input_gen_seed, num_inputs)
-            for j, input_ in enumerate(inputs):
-                input_.save(f"{test_case_dir}/input{j}.bin")
+                             "Use --permit-overwrite to overwrite the test case")
+
+            # set the seed and generate the program. After each generated
+            # program, we'll increase the seed by one
```

Comment:
Input and program generators already take care of it, no need to duplicate it here